### ISSUE TITLE
v1.0.2b - 'mountvol' is not recognized as an internal or external command, operable program or batch file.

### ISSUE THREAD

**Issue by rsmith16384** [cisagov/CHIRP#22]
# 🐛 Summary #

What's wrong? Please be specific.

When running the python code, this error is immediately displayed and appears to affect subsequent operations (scan appears to hang):

C:\ARTemp\chirp\LATEST\CHIRP-main>C:\Users\rsmit\AppData\Local\Programs\Python\Python39-32\python.exe chirp.py
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│ C:\ARTemp\chirp\LATEST\CHIRP-main\chirp.py:16 in 'mountvol' is not recognized as an internal or external command,
operable program or batch file.
<module>'mountvol' is not recognized as an internal or external command,
operable program or batch file.
                                        │
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
│                                                                                                  │'mountvol' is not recognized as an internal or external command,
operable program or batch file.

│   13 if __name__ == "__main__":                                                                  │
│   14 │   try:                                                                                    │
│   15 │   │   freeze_support()                                                                    │
│ > 16 │   │   run.run()                                                                           │
│   17 │   │   time.sleep(2)                                                                       │
│   18 │   │   CONSOLE(                                                                            │
│   19 │   │   │   "[green][+][/green] DONE! Your results can be found in {}. Press any key to     │
│                                                                                                  │
│ C:\ARTemp\chirp\LATEST\CHIRP-main\chirp\run.py:19 in run                                         │
│                                                                                                  │
│   16 │   if not os.path.exists(OUTPUT_DIR):                                                      │
│   17 │   │   os.mkdir(OUTPUT_DIR)                                                                │
│   18 │   plugins = loader.load()                                                                 │
│ > 19 │   run_plugins(plugins)                                                                    │
│   20                                                                                             │
│   21                                                                                             │
│   22 def run_plugins(plugins: Dict[str, Callable]) -> None:                                      │
│                                                                                                  │
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
│ C:\ARTemp\chirp\LATEST\CHIRP-main\chirp\run.py:29 in run_plugins                                 │
│                                                                                                  │
│   26 │   'mountvol' is not recognized as an internal or external command,
operable program or batch file.
:type plugins: Dict[str, Callable]                                                      │
│   27 │   """                                                                                     │
│   28 │   _loop = asyncio.get_event_loop()                                                        │
│ > 29 │   _loop.run_until_complete(_run_coroutines(plugins))                                      │
│   30                                                                                             │
│   31                                                                                             │
│   32 async def _run_coroutines(plugins: Dict[str, Callable]) -> None:                            │
│                                                                                                  │
│ C:\Users\rsmit\AppData\Local\Programs\Python\Python39-32\lib\asyncio\base_events.py'mountvol' is not recognized as an internal or external command,
operable program or batch file.
:642 in       │
│ run_until_complete                                                                               │
│                                                                                                  │
│    639 │   │   if not future.done():                                                             │
│    640 │   │   │   raise RuntimeError('Event loop stopped before Future completed.')             │
│    641 │   │                                                                                     │
'mountvol' is not recognized as an internal or external command,
operable program or batch file.
│ >  642 │   │   return future.result()                                                            │
│    643 │                                                                                         │
│    644 │   def stop(self):                                                                       │
│    645 │   │   """Stop running the event loop.                                                   │
│                                                                                                  │
│ C:\ARTemp\chirp\LATEST\CHIRP-main\chirp\run.py:43 in _run_coroutines                             │
│                                                                                                  │
│   40 │   │   │   load.from_yaml(get_indicators()), list(plugins.keys())                          │
│   41 │   │   )                                                                                   │
│   42 │   )                                                                                       │
│ > 43 │   await asyncio.gather(                                                                   │
│   44 │   │   *[                                                                                  │
│   45 │   │   │   entrypoint(                                                                     │
│   46 │   │   │   │   [                                                                           │
│                                                                                                  │
│ C:\ARTemp\chirp\LATEST\CHIRP-main\chirp\plugins\events\scan.py:129 in run                        │
│                                                                                                  │
│   126 │   async with aiomp.Pool() as pool:                                                       │
│   127 │   │   try:                                                                               │
│   128 │   │   │   async for i in pool.map(_run, tuple(run_args)):                                │
│ > 129 │   │   │   │   _rep = i[0]                                                                │
│   130 │   │   │   │   num_logs += i[1]                                                           │
│   131 │   │   │   │   for k, v in _rep.items():                                                  │
│   132 │   │   │   │   │   try:                                                                   │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
TypeError: 'NoneType' object is not subscriptable
[07:36:38] [!] We can't find windows event logs at their standard path.                                   common.py:104
           [EVENTS] Entered events plugin.                                                                common.py:104
           [NETWORK] Entered network plugin.                                                              common.py:104
           [NETWORK] Read 163 records, found 0 IoC hits.                                                  common.py:104
           [REGISTRY] Entered registry plugin.                                                            common.py:104
           [REGISTRY] Reading HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\sibot          common.py:104
           [REGISTRY] Key HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\sibot does not     common.py:104
           exist.
           [REGISTRY] Reading HKEY_LOCAL_MACHINE\Software\Microsoft\CTF                                   common.py:104
           [REGISTRY] Reading HKLM\SOFTWARE\Wow6432Node\Microsoft\Windows NT\CurrentVersion\Image File    common.py:104
           Execution Options\
           [REGISTRY] Found 0 hit(s) for Sibot - Registry indicator.                                      common.py:104
           [REGISTRY] Found 0 hit(s) for Teardrop - Registry Activity indicator.                          common.py:104
           [REGISTRY] Found 0 hit(s) for IFEO Persistence indicator.                                      common.py:104
           [YARA] Entered yara plugin.                                                                    common.py:104
           [YARA] Enumerating the entire filesystem due to ['CISA Solar Fire', 'CISA Teardrop',           common.py:104
           'CrowdStrike Rempack', 'CrowdStrike Sunspot', 'FireEye Cosmic Gale', 'FireEye Sunburst']...
           this is going to take a while.
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104
           [YARA] Beginning processing.                                                                   common.py:104
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104
[07:36:40] [[07:36:40]! ] We can't find windows event logs at their standard path.                                   [common.py!:104
] We can't find windows event logs at their standard path.                                   common.py:104
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104
[07:36:40] [!] We can't find windows event logs at their standard path.                                   common.py:104

## To reproduce ##

Download latest code
   Install with python.exe -m pip install -e .
   Run with python.exe chirp.py
   Observe output

## Expected behavior ##

No mountvol error

## Any helpful log output or screenshots ##

Paste the results here:

![mountvol](https://user-images.githubusercontent.com/37616409/112165331-e1450f00-8bab-11eb-8635-5a79ced62137.png)

Add any screenshots of the problem here.

**Comment by DeemOnSecurity** [comment]
What version of Windows is this running on? `mountvol` is a Windows command line tool.

**Comment by rsmith16384** [comment]
![winver2](https://user-images.githubusercontent.com/37616409/112344202-1b341500-8c81-11eb-9dbc-c0096a8399da.png)

**Comment by DaleEMoore** [comment]
I have the same problem on Ubuntu 20.04 Linux:

```
dalem@goLangNFyne:/media/vmShare/SALB/v1.0.3/1/CHIRP-1.0.3$ !278
python3 chirp.py 
/bin/sh: 1: mountvol: not found
/bin/sh: 1: mountvol: not found
[08:13:37] [YARA] Entered yara plugin.                             common.py:108
           [YARA] Enumerating the entire filesystem due to         common.py:108
           ['CrowdStrike Rempack', 'CrowdStrike Sunspot', 'FireEye              
           Cosmic Gale', 'CISA Solar Fire', 'FireEye Sunburst',                 
           'CISA Teardrop']... this is going to take a while.                   
/bin/sh: 1: mountvol: not found
[08:13:39] [YARA] Done. Processed 1 files.                         common.py:108
           [YARA] Found 0 hit(s) for yara indicators.              common.py:108
[08:13:41] [+] DONE! Your results can be found in                  common.py:108
           /media/vmShare/SALB/v1.0.3/1/CHIRP-1.0.3/output. Press               
           any key to exit. 

```
Perhaps I have overlooked some Ubuntu 20.04 Linux installation procedure?

Any suggestions you have are very much appreciated!

**Comment by DeemOnSecurity** [comment]
> I have the same problem on Ubuntu 20.04 Linux:
> 
> ```
> dalem@goLangNFyne:/media/vmShare/SALB/v1.0.3/1/CHIRP-1.0.3$ !278
> python3 chirp.py 
> /bin/sh: 1: mountvol: not found
> /bin/sh: 1: mountvol: not found
> [08:13:37] [YARA] Entered yara plugin.                             common.py:108
>            [YARA] Enumerating the entire filesystem due to         common.py:108
>            ['CrowdStrike Rempack', 'CrowdStrike Sunspot', 'FireEye              
>            Cosmic Gale', 'CISA Solar Fire', 'FireEye Sunburst',                 
>            'CISA Teardrop']... this is going to take a while.                   
> /bin/sh: 1: mountvol: not found
> [08:13:39] [YARA] Done. Processed 1 files.                         common.py:108
>            [YARA] Found 0 hit(s) for yara indicators.              common.py:108
> [08:13:41] [+] DONE! Your results can be found in                  common.py:108
>            /media/vmShare/SALB/v1.0.3/1/CHIRP-1.0.3/output. Press               
>            any key to exit. 
> ```
> 
> Perhaps I have overlooked some Ubuntu 20.04 Linux installation procedure?
> 
> Any suggestions you have are very much appreciated!

Ah, that is likely an if statement we need to implement, as we shouldn't be trying to run `mountvol` (a Windows binary) on Linux.

**Comment by DeemOnSecurity** [comment]
> ![winver2](https://user-images.githubusercontent.com/37616409/112344202-1b341500-8c81-11eb-9dbc-c0096a8399da.png)

This is perplexing as `mountvol` was added to Windows in Windows 7/2008. I suppose I can find a better workaround for that bit of code.

**Comment by lindacorbett** [comment]
So is bug fixer getting fixed

**Comment by DeemOnSecurity** [comment]
> So is bug fixer getting fixed

This bug is fixed with the push of v1.0.4, which is downloadable from [Releases](https://github.com/cisagov/CHIRP/releases)
